File: Sources/MutsumiPet/Models/PetPose.swift
enum PetPose: String, CaseIterable, Hashable {
    case idle
    case curious
    case happy
    case sleeping
    case grabbed

    var assetName: String {
        switch self {
        case .idle: "mutsumi_pet"
        case .curious: "mutsumi_curious"
        case .happy: "mutsumi_happy"
        case .sleeping: "mutsumi_sleeping"
        case .grabbed: "mutsumi_grabbed"
        }
    }
}

File: windows/MutsumiPet.Windows/Models/PetPose.cs
namespace MutsumiPet.Models
{
    public enum PetPose
    {
        Idle,
        Curious,
        Happy,
        Sleeping,
        Grabbed
    }

    public static class PetPoses
    {
        public static readonly PetPose[] AllCases =
        {
            PetPose.Idle,
            PetPose.Curious,
            PetPose.Happy,
            PetPose.Sleeping,
            PetPose.Grabbed
        };

        public static string AssetName(PetPose pose)
        {
            switch (pose)
            {
                case PetPose.Curious: return "mutsumi_curious";
                case PetPose.Happy: return "mutsumi_happy";
                case PetPose.Sleeping: return "mutsumi_sleeping";
                case PetPose.Grabbed: return "mutsumi_grabbed";
                default: return "mutsumi_pet";
            }
        }
    }
}
